src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -42,35 +45,133 @@ public void editPerson() {
+        assertEquals(alicePersonCard, newAlice);
 
         //Confirm the underlying person object has the right values
         assertEquals(newAlice.toString(), personListPanel.getSelectedPerson().toString());
 
-        //TODO: confirm again after the next sync
-        //TODO: confirm other cards are unaffected
-        //TODO: confirm the right card is selected after the edit
-        //TODO: confirm status bar is updated correctly
 
+        //Confirm again after the next sync
+        mainGui.sleep(getTestingConfig().getUpdateInterval(), TimeUnit.MILLISECONDS);
+        assertEquals(newAlice.toString(), personListPanel.getSelectedPerson().toString());
+
+        //Confirm other cards are unaffected.
+        personListPanel.clickOnListView();
+        personListPanel.use_LIST_JUMP_TO_INDEX_SHORTCUT(2); //To make sure the card is visible
+        assertEquals(personListPanel.getPersonCardHandle(1), td.benson); //Ensure the card details are correct
+        personListPanel.use_LIST_JUMP_TO_INDEX_SHORTCUT(3);
+        assertEquals(personListPanel.getPersonCardHandle(2), td.charlie);
+        personListPanel.use_LIST_JUMP_TO_INDEX_SHORTCUT(4);
+        assertEquals(personListPanel.getPersonCardHandle(3), td.dan);
+
+        //Confirm status bar is updated correctly
+        assertEquals(statusBar.getText(), "Edit Person [ Alice Brown -> Alicia Brownstone ] completed successfully.");
+    }
+
+
+    @Test
+    public void editPerson_usingContextMenu() {
+        personListPanel.rightClickOnPerson(td.alice);
+        EditPersonDialogHandle editPersonDialog = personListPanel.clickOnContextMenu(
+                PersonListPanelHandle.ContextMenuChoice.EDIT);
+        assertTrue(editPersonDialog.isValidEditDialog());
+    }
+
+
+    @Test
+    public void editPerson_usingEditButton() {
+        personListPanel.clickOnPerson(td.alice);
+        EditPersonDialogHandle editPersonDialog =  personListPanel.clickEdit();
+        assertTrue(editPersonDialog.isValidEditDialog());
+    }
+
+    @Test
+    public void editPerson_dataValidation() {
+
+        personListPanel.clickOnPerson(td.alice);
+        EditPersonDialogHandle editPersonDialog =  personListPanel.clickEdit();
+        editPersonDialog.enterFirstName("Peter");
+        editPersonDialog.enterLastName("");
+        editPersonDialog.clickOk();
+
+        assertTrue(editPersonDialog.isInputValidationErrorDialogShown());
+        editPersonDialog.dissmissErrorMessage("Invalid Fields");
+        assertFalse(editPersonDialog.isInputValidationErrorDialogShown());
+    }
+
+    @Test
+    public void cancelOperation_usingAccelerator() {
+
+        //Delete
+        personListPanel.clickOnPerson(td.alice);
+        PersonCardHandle deletedCard = personListPanel.getPersonCardHandle(new Person(personListPanel.getSelectedPerson()));
+        personListPanel.use_PERSON_DELETE_ACCELERATOR();
+        mainGui.sleep(1, TimeUnit.SECONDS);
+        assertTrue(deletedCard.isShowingGracePeriod("Deleting"));
```

Comment:
The sleep can be inside the `isShowingGracePeriod`? Otherwise you have to repeat the sleep every time. When you find yourself repeating a code block, it's as sign that you are missing a higher level abstraction.